api/resources_portal/management/commands/material_additional_metadata_key_change.py

```diff
@@ -0,0 +1,119 @@
+
+
+def addKey(category, new_key):
+    querySet = None
+    materials_changed = 0
+
+    if category:
+        if not validateCategory(category):
+            logger.error(f"The provided category '{category}' is not a valid material category.")
+            return
+        querySet = Material.objects.filter(category=category)
+    else:
+        querySet = Material.objects.all()
+
+    with transaction.atomic():
+        for material in querySet:
+            material.additional_metadata[new_key] = ""
+            material.save()
+            materials_changed += 1
+
+    logger.info(f"The key '{new_key}' was added to {materials_changed} materials.")
+
+
+def removeKey(category, old_key):
+    querySet = None
+    materials_changed = 0
+
+    if category:
+        if not validateCategory(category):
+            logger.error(f"The provided category '{category}' is not a valid material category.")
+            return
+        querySet = Material.objects.filter(category=category)
+    else:
+        querySet = Material.objects.all()
+
+    with transaction.atomic():
+        for material in querySet:
+            material.additional_metadata.pop(old_key)
+            material.save()
+            materials_changed += 1
+
+    logger.info(f"The key '{old_key}' was removed from {materials_changed} materials.")
+
+
+def material_additional_metadata_key_change(category, old_key, new_key):
+    querySet = None
+    materials_changed = 0
+
+    if category:
+        if not validateCategory(category):
+            logger.error(f"The provided category '{category}' is not a valid material category.")
+            return
+        querySet = Material.objects.filter(additional_metadata__has_key=old_key, category=category)
+    else:
+        querySet = Material.objects.filter(additional_metadata__has_key=old_key)
+
+    # Note that this doesn't use the above add and remove commands. This is because doing so would require twice the queries.
+    with transaction.atomic():
+        for material in querySet:
+            material.additional_metadata[new_key] = material.additional_metadata[old_key]
+            material.additional_metadata.pop(old_key)
+            material.save()
+            materials_changed += 1
+
+    logger.info(f"The key '{old_key}' was changed to '{new_key}' in {materials_changed} materials.")
+
+
+# This is a command to add, remove, or change the names of additional metadata fields currently in the server.
+# It has the following use cases:
+# 1. Change a metadata key
+# ./rportal migrate-metadata-keys old_key new_key --category [CATEGORY]
+# 2. Add a metadata key
+# ./rportal migrate-metadata-keys --add new_key --category [CATEGORY]
+# 2. Remove a metadata key
+# ./rportal migrate-metadata-keys --remove old_key --category [CATEGORY]
+# For each use case, the category parameter is optional, and categories should be specified in all caps, like "PLASMID" or "MODEL_ORGANISM".
```

Comment:
This comment would make a great docstring for the class.

Reply:
Good point. Done!